service/api/src/test/java/org/eclipse/kapua/KapuaIllegalArgumentExceptionTest.java

```diff
@@ -0,0 +1,126 @@
+                assertEquals("Expected and actual values should be the same.", "An illegal value was provided for the argument " + name + ": " + value + ".", kapuaIllegalArgumentException.getMessage());
+                assertNull("Null expected.", kapuaIllegalArgumentException.getCause());
+            }
+        }
+    }
+
+    @Test
+    public void kapuaIllegalArgumentExceptionKapuaErrorCodesStringParametersTest() {
+        for (String name : argumentName) {
+            for (String value : argumentValue) {
+                String[] message = {"The entity of type " + name + " with id/name " + value + " was not found.", "Error: " + name + ", " + value, "An entity with the same name " + name + " already exists.",
+                        "An entity with the same external Id " + name + " already exists.", "Error: " + name + ", " + value, "The current subject is not authorized for " + name + ".", "An illegal value was provided for the argument " + name + ": " + value + ".",
+                        "An illegal null value was provided for the argument " + name + ".", "The application is in a illegal state: " + name + ".", "The entity is out of state as it has been modified or deleted by another transaction.",
+                        "No authenticated Subject found in context.", "Error: " + name + ", " + value, "An internal error occurred: " + name + ".", "Error: " + name + ", " + value, "Maximum number of items has been exceeded,",
+                        "User does not have permission to perform this action. Missing permission: " + name + ". Please perform a new login to refresh users permissions.", "An entity with the same name " + name,
+                        "An entity with the same external id " + name + " already exists in another account.", "Bundle could not be started. Please check the device log for errors.", "Bundle could not be stopped. Please check the device log for errors.",
+                        "Error: " + name + ", " + value, "Max number of " + name + " reached. Please increase the number or set InfiniteChild" + name + " parameter to True.", "The start time cannot be later than the end time.",
+                        "Start and end time cannot be at the same point in time. Please change the parameters.", "There should only be one scheduling parameter - either Retry interval or Cron schedule.",
+                        "The selected devices were not found. Please refresh device list.", "An entity with the same value for field already exists here or in another job.",
+                        "Error: Another resource is currently downloading. Please wait a few moments so package finishes downloading and then try again.",
+                        "Error: " + name + ", " + value, "Operation not allowed on admin role.", "Operation not allowed on this specific permission.", "The Service is disabled: " + name};
+
+                for (int i = 0; i < kapuaErrorCodes.length; i++) {
```

Comment:
Please use enhanced for loop: "                for (KapuaErrorCodes ker: kapuaErrorCodes) {"

Reply:
I used index i to get values from different arrays in the same position